src/client/hud.cpp

```diff
@@ -567,7 +652,21 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 				client->getMinimap()->drawMinimap(rect);
 				break; }
 			case HUD_ELEM_HOTBAR: {
-				drawHotbar(pos, e->offset, e->dir, e->align);
+				// Handle splitting caused by hud_hotbar_max_width
+				u16 hotbar_itemcount = player->hotbar_source.getMaxLength();
+				s32 width = hotbar_itemcount * (m_hotbar_imagesize + m_padding * 2);
+				const v2u32 &window_size = RenderingEngine::getWindowSize();
+				if ((float) width / (float) window_size.X >
+						g_settings->getFloat("hud_hotbar_max_width")) {
+					v2s32 upper_pos = pos - v2s32(0, m_hotbar_imagesize + m_padding);
+					u16 upper_itemcount = hotbar_itemcount/2;
+					drawHotbar(upper_pos, e->offset, e->dir, e->align, upper_itemcount, 0);
+					drawHotbar(pos, e->offset, e->dir, e->align,
+							hotbar_itemcount - upper_itemcount, upper_itemcount);
+					break;
```

Comment:
Superfluous `break;`. Could we perhaps use (almost) the same code in both cases by using different values for `upper_itemcount` instead? `drawHotbar` could return early if there's 0 slots to draw. As of now this code is not very pleasant to read.